### PR TITLE
Supports column formatter options returns dom element or jquery element

### PR DESCRIPTION
**Pull request by weibangtuo**
**🤔Type of Request**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**

**📝Changelog**

- [x] **Core**
- [ ] **Extensions**

**💡Example(s)?**

[Example online](https://live.bootstrap-table.com/code/weibangtuo/18018)

Code example
```js
$('#table').bootstrapTable({
  columns: [
    {
      field: 'html',
      title: 'HTML',
      formatter: val => {
        const div = document.createElement('div')
  
        div.append('html')
        div.addEventListener('click', () => { alert('From html') })
        return div
      }
    },
    {
      field: 'jquery',
      title: 'jQuery',
      formatter: val => {
        const div = $('<div>')
  
        div.html('jquery')
        div.on('click', () => { alert('From jquery') })
        return div
      }
    },
    {
      field: 'vue3',
      title: 'Vue3 Component',
      formatter: val => {
        // import { h, render, getCurrentInstance } from 'vue'
        // import MyComponent from './MyComponent.vue'

        const vnode = h(MyComponent, props, children)
        const app = this.$.appContext // or const app = getCurrentInstance().appContext.app
        const div = document.createElement('div')

        vnode.appContext = app
        render(vnode, div)
        return div
      }
    },
    {
      field: 'echart',
      title: 'EChart Example',
      formatter: val => {
        const chartDom = document.createElement('div')

        // Size needs to be set because DOM is not mounted immediate and size cannot be obtained.
        chartDom.style.width = '300px'
        chartDom.style.height = '200px'

        const myChart = echarts.init(chartDom)
  
        myChart.setOption({
          xAxis: {
            type: 'category',
            data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
          },
          yAxis: {
            type: 'value'
          },
          series: [
            {
              data: [150, 230, 224, 218, 135, 147, 260],
              type: 'line'
            }
          ]
        })
        return chartDom
      }
    },
  ]
})
```

**☑️Self Check before Merge**

⚠️ Please check all items below before reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by wenzhixin** [comment]
* 可以使用 https://live.bootstrap-table.com/ 提供在线例子，写对应的分支即可。
* 文档应该需要更新下，把场景和支持的例子加上。

**Comment by wenzhixin** [comment]
vue + develop example: https://live.bootstrap-table.com/code/wenzhixin/18014

**Comment by weibangtuo** [comment]
Updated, Examle Online: https://live.bootstrap-table.com/code/weibangtuo/18018

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1615,46 +1586,32 @@ class BootstrapTable {
       }
 
       // Style concat
-      if (csses.concat([this.header.styles[j]]).length) {
+      if (csses.length || this.header.styles[j].length) {
         styleToAdd_ += `${csses.concat([this.header.styles[j]]).join('; ')}`
       }
       if (item[`_${field}_style`]) {
         styleToAdd_ += `${item[`_${field}_style`]}`
       }
```
Comment (wenzhixin) [inline src/bootstrap-table.js:1594]: 上面的 `csses` 考虑放到 `tr` 中，`td` 处理为：`this.header.styles[j]` + `item[`_${field}_style`]` + 下面的 `cellStyle.css`

File: site/docs/api/column-options.md [hunk 2]
```diff
@@ -230,7 +230,7 @@ The column options is defined in `jQuery.fn.bootstrapTable.columnDefaults`.
   * `data`: Array of all the data rows.
   * `value`: If footer data is set, the value of the footer column.
 
-  The function should return a string with the text to show in the footer cell.
+  Expected return data type: `jQuery`, `String`, `HTMLElement`. Other types will be forced to `String` type.
```
Comment (wenzhixin) [inline site/docs/api/column-options.md:233]:
```suggestion
  The expected return data type is `jQuery`, `String`, or `HTMLElement`. Other types will be forced to the `String` type.
```

File: site/docs/api/column-options.md [hunk 3]
```diff
@@ -282,6 +282,8 @@ The column options is defined in `jQuery.fn.bootstrapTable.columnDefaults`.
   * `index`: the row index.
   * `field`: the row field.
 
+  Expected return data type: `jQuery`, `String`, `HTMLElement`. Other types will be forced to `String` type.
```
Comment (wenzhixin) [inline site/docs/api/column-options.md:285]:
```suggestion
  The expected return data type is `jQuery`, `String`, or `HTMLElement`. Other types will be forced to the `String` type.
```

**Comment by wenzhixin** [comment]
这个例子原来的写法失效了，帮忙修改下：
https://live.bootstrap-table.com/code/wenzhixin/18027

**Review by wenzhixin — APPROVED** [review]

**Comment by wenzhixin** [comment]
Refs:
* https://github.com/wenzhixin/bootstrap-table/pull/7515
* https://github.com/wenzhixin/bootstrap-table/pull/7521
* https://github.com/wenzhixin/bootstrap-table/pull/7571
* https://github.com/wenzhixin/bootstrap-table/pull/7584
* https://github.com/wenzhixin/bootstrap-table-examples/pull/537

**Pull request merged**